!! Do not merge!!

sbeaver-netgate opened the pull request:
Partially converted to bootstrap.

Code rearranged and commented
Indenting made consistent
Fixed issue with adding aliases to a config file where noe yet exist
Layout changed to look more like other bootstrap pages (I hope)

sbeaver-netgate committed:
!! Do not merge!!

Partially converted to bootstap.

Code rearranged and commented
Indenting made consistent
Fixed issue with adding aliases to a config file where noe yet exist
Layout changed to look more like other bootstrap pages (I hope)

SjonHortensius commented on usr/local/www/diag_dns.php:
```diff
@@ -29,20 +29,24 @@
-	$a_aliases = array();
+/* If this section of config.xml has not been populated yet we need to set it up 
+*/
+if (!is_array($config['aliases']['alias'])) {
```
Please change only the layout, and logic if necessary. This specifically isn't necessary and only diffuses the diff

sbeaver-netgate replied:
You mean don’t add comments?

That logic change and comment were added upstream to the pfSense master.

Steve

> On Mar 26, 2015, at 4:50 PM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/diag_dns.php https://github.com/SjonHortensius/pfsense/pull/42#discussion_r27254529:
> 
> >  require("guiconfig.inc");
> > 
> >  $host = trim($_REQUEST['host'], " \t\n\r\0\x0B[];\"'");
> >  $host_esc = escapeshellarg($host);
> > 
> > -if (is_array($config['aliases']['alias'])) {
> > -   $a_aliases = &$config['aliases']['alias'];
> >   -} else {
> > -   $a_aliases = array();
> >   +/\* If this section of config.xml has not been populated yet we need to set it up 
> >   +*/
> >   +if (!is_array($config['aliases']['alias'])) {
> >   Please change only the layout, and logic if necessary. This specifically isn't necessary and only diffuses the diff
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/42/files#r27254529.

SjonHortensius replied:
Then please, revert it. Base your changes on this repo and let me merge from upstream now and then. It'll keep things clear. I only want bootstrap related changes in this repo :)

sbeaver-netgate replied:
I can do that, but I am concerned that the page does not work as required without this bug-fix and so will not pass unit testing.

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com | www.netgate.com 

> On Mar 28, 2015, at 7:54 AM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/diag_dns.php https://github.com/SjonHortensius/pfsense/pull/42#discussion_r27344508:
> 
> >  require("guiconfig.inc");
> > 
> >  $host = trim($_REQUEST['host'], " \t\n\r\0\x0B[];\"'");
> >  $host_esc = escapeshellarg($host);
> > 
> > -if (is_array($config['aliases']['alias'])) {
> > -   $a_aliases = &$config['aliases']['alias'];
> >   -} else {
> > -   $a_aliases = array();
> >   +/\* If this section of config.xml has not been populated yet we need to set it up 
> >   +*/
> >   +if (!is_array($config['aliases']['alias'])) {
> >   Then please, revert it. Base your changes on this repo and let me merge from upstream now and then. It'll keep things clear. I only want bootstrap related changes in this repo :)
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/42/files#r27344508.

SjonHortensius commented on usr/local/www/diag_dns.php:
```diff
@@ -170,116 +174,142 @@ function display_host_results ($address,$hostname,$dns_speeds) {
+<!-- First table displayes the DNS lookup form and hte resulting IP addresses -->
+
+	<div class="table-responsive">
+		<table width="100%" class="table" summary="results">
```
Width & summary should not be included I think

SjonHortensius commented on usr/local/www/diag_dns.php:
```diff
@@ -170,116 +174,142 @@ function display_host_results ($address,$hostname,$dns_speeds) {
+		<table width="100%" class="table" summary="results">
+		<thead>
+			<tr>
+				<th width="30%">Host name</th>
```
Is this width attribute necessary?

sbeaver-netgate replied:
Probably not. I’ll remove that stuff when I re-do the form tomorrow.

> On Mar 26, 2015, at 4:52 PM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/diag_dns.php https://github.com/SjonHortensius/pfsense/pull/42#discussion_r27254704:
> 
> > -            <?=$mandfldhtml;?>
> > -           <table summary="results">
> > -               <tr><td valign="top">
> > -           <input name="host" type="text" class="formfld" id="host" size="20" value="<?=htmlspecialchars($host);?>" />
> > -           </td>
> > -           <?php if ($resolved && $type) { ?>
> > -           <td valign="middle">&nbsp;=&nbsp;</td><td>
> > -           <font size="+1">
> >   +
> >   +<!-- First table displayes the DNS lookup form and hte resulting IP addresses -->
> >   +
> > -   <div class="table-responsive">
> > -       <table width="100%" class="table" summary="results">
> > -       <thead>
> > -           <tr>
> > -               <th width="30%">Host name</th>
> >   Is this width attribute necessary?
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/42/files#r27254704.

SjonHortensius commented on usr/local/www/diag_dns.php:
```diff
@@ -170,116 +174,142 @@ function display_host_results ($address,$hostname,$dns_speeds) {
+		</thead>
+		<tbody>
+			<tr class="active">
+				<td valign="top">
```
valign definetly should not be here

sbeaver-netgate replied:
Sure. I have not cleaned up the old code yet. Just looking for comments on the screen appearance and functionality at this point.

> On Mar 26, 2015, at 4:53 PM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/diag_dns.php https://github.com/SjonHortensius/pfsense/pull/42#discussion_r27254844:
> 
> > +
> > +<!-- First table displayes the DNS lookup form and hte resulting IP addresses -->
> > +
> > -   <div class="table-responsive">
> > -       <table width="100%" class="table" summary="results">
> > -       <thead>
> > -           <tr>
> > -               <th width="30%">Host name</th>
> > -               <?php if ($resolved && $type) { ?> 
> > -               <th>IP Addresse(s)</th>
> > -               <?php } ?>
> > -           </tr>
> > -       </thead>
> > -       <tbody>
> > -           <tr class="active">
> > -               <td valign="top">
> >   valign definetly should not be here
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/42/files#r27254844.

SjonHortensius commented on usr/local/www/diag_dns.php:
```diff
@@ -170,116 +174,142 @@ function display_host_results ($address,$hostname,$dns_speeds) {
-</td></tr></table>
-<?php include("fend.inc"); ?>
+<?php include("foot.inc"); ?>
 </body>
```
Foot.inc includes `</body></html>` so those can be removed

sbeaver-netgate replied:
OK thanks.

> On Mar 26, 2015, at 4:56 PM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/diag_dns.php https://github.com/SjonHortensius/pfsense/pull/42#discussion_r27255192:
> 
> >  </form>
> > -</td></tr></table>
> > -<?php include("fend.inc"); ?>
> > +<?php include("foot.inc"); ?>
> >  </body>
> > Foot.inc includes </body></html> so those can be removed
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/42/files#r27255192.

sbeaver-netgate committed:
print_input_errors() is now a wrapper around print_errors().
print_errors() allows the caller to supply the error messages and the alert box heading.

sbeaver-netgate committed:
Completed Bootstrap conversion
Added error message on lookup failure

SjonHortensius commented on usr/local/www/diag_dns.php:
```diff
@@ -91,7 +95,7 @@
 if ($_POST) {
 	unset($input_errors);
-
+	
```
Please configure your IDE to strip trailing whitespace, otherwise we'd have to keep running clean.sh on all commits ;)

SjonHortensius commented on usr/local/www/diag_dns.php:
```diff
@@ -170,116 +171,150 @@ function display_host_results ($address,$hostname,$dns_speeds) {
+	}
+?>
+  
+<form action="diag_dns.php" method="post" name="iform" id="iform">
```
Please read https://github.com/SjonHortensius/pfsense#forms. We're replacing _all_ forms by Form classes

SjonHortensius commented on usr/local/www/guiconfig.inc:
```diff
@@ -266,11 +266,15 @@ function do_input_validation($postdata, $reqdfields, $reqdfieldsn, &$input_error
 }
 
 function print_input_errors($input_errors) {
+    print_errors($input_errors, gettext('The following input errors were detected:'));
```
I don't see any reason for refactoring this, it didn't come from master either?

sbeaver-netgate replied:
I wanted a function to display a general error message. This seemed like the most efficient way to do that. What would you suggest?

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com | www.netgate.com 

> On Mar 28, 2015, at 10:50 AM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/guiconfig.inc https://github.com/SjonHortensius/pfsense/pull/42#discussion_r27345559:
> 
> > @@ -266,11 +266,15 @@ function do_input_validation($postdata, $reqdfields, $reqdfieldsn, &$input_error
> >  }
> > 
> >  function print_input_errors($input_errors) {
> > -    print_errors($input_errors, gettext('The following input errors were detected:'));
> >   I don't see any reason for refactoring this, it didn't come from master either?
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/42/files#r27345559.

SjonHortensius replied:
You are outputting 1 error about 1 host on 1 page. How does this warrant a generic method? Looking at the code a lot of people like doing it that way @ pfsense but I don't like it. You create a generic function after you have more then 1 usecase for it. Only then can you know what your parameters will be.

For now, I'd simply output html in diag_dns; something like `<div class="alert alert-warning" role="alert"><?=htmlspecialchars($host)?> could not be resolved</div>`

sbeaver-netgate replied:
It seems likely a generic error message will b required on most pages?

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com | www.netgate.com 

> On Mar 28, 2015, at 11:12 AM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/guiconfig.inc https://github.com/SjonHortensius/pfsense/pull/42#discussion_r27345671:
> 
> > @@ -266,11 +266,15 @@ function do_input_validation($postdata, $reqdfields, $reqdfieldsn, &$input_error
> >  }
> > 
> >  function print_input_errors($input_errors) {
> > -    print_errors($input_errors, gettext('The following input errors were detected:'));
> >   You are outputting 1 error about 1 host on 1 page. How does this warrant a generic method? Looking at the code a lot of people like doing it that way @ pfsense but I don't like it. You create a generic function after you have more then 1 usecase for it. Only then can you know what your parameters will be.
> 
> For now, I'd simply output html in diag_dns; something like <div class="alert alert-warning" role="alert"><?=htmlspecialchars($host)?> could not be resolved</div>
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/42/files#r27345671.

SjonHortensius replied:
What's wrong with writing the HTML directly?

sbeaver-netgate replied:
Nothing. I was just following the technique used in the existing code. 

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com | www.netgate.com 

> On Mar 28, 2015, at 11:19 AM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/guiconfig.inc https://github.com/SjonHortensius/pfsense/pull/42#discussion_r27345716:
> 
> > @@ -266,11 +266,15 @@ function do_input_validation($postdata, $reqdfields, $reqdfieldsn, &$input_error
> >  }
> > 
> >  function print_input_errors($input_errors) {
> > -    print_errors($input_errors, gettext('The following input errors were detected:'));
> >   What's wrong with writing the HTML directly?
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/42/files#r27345716.

SjonHortensius commented on usr/local/www/diag_dns.php:
```diff
@@ -170,116 +171,150 @@ function display_host_results ($address,$hostname,$dns_speeds) {
+							
+				<td>
+					<?php if ($resolved && $type) { ?>
+					<font size="+1">
```
you should probably rewrite this to a `TH`, `<font>` is deprecated

SjonHortensius commented on usr/local/www/diag_dns.php:
```diff
@@ -170,116 +171,150 @@ function display_host_results ($address,$hostname,$dns_speeds) {
+			</tr>
+		</thead>
+		<tbody>
+			<tr class="active">
```
Pretty sure tr.active doesn't match any css; what should it do?

sbeaver-netgate replied:
"table table-striped" makes rows alternate dark/light.

“active" overrides that and sets the particular row to the darker shade. So if you choose not to specify "table-striped" it allows you to control which rows have the darker shading.

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com | www.netgate.com 

> On Mar 28, 2015, at 10:56 AM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/diag_dns.php https://github.com/SjonHortensius/pfsense/pull/42#discussion_r27345597:
> 
> > +<form action="diag_dns.php" method="post" name="iform" id="iform">
> > +
> > +<!-- First table displayes the DNS lookup form and the resulting IP addresses -->
> > -   <div class="table-responsive">
> > -   
> > -       <table class="table" summary="results">
> > -       <thead>
> > -           <tr>
> > -               <th width="30%">Host name</th>
> > -               <?php if ($resolved && $type) { ?> 
> > -               <th>IP Addresse(s)</th>
> > -               <?php } ?>
> > -           </tr>
> > -       </thead>
> > -       <tbody>
> > -           <tr class="active">
> >   Pretty sure tr.active doesn't match any css; what should it do?
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/42/files#r27345597.

SjonHortensius commented on usr/local/www/diag_dns.php:
```diff
@@ -170,116 +171,150 @@ function display_host_results ($address,$hostname,$dns_speeds) {
+<form action="diag_dns.php" method="post" name="iform" id="iform">
+
+<!-- First table displayes the DNS lookup form and the resulting IP addresses -->
+	<div class="table-responsive">
```
I think this setup should be refactored; we don't need tables for layout. I think you should use panels for this; have a look at http://getbootstrap.com/components/#panels

sbeaver-netgate replied:
OK

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com | www.netgate.com 

> On Mar 28, 2015, at 11:15 AM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/diag_dns.php https://github.com/SjonHortensius/pfsense/pull/42#discussion_r27345696:
> 
> > -                           echo $hostitem . "<br />";
> > -                           $found++;
> > -   /\* Dosplay any error messages resulting from user input */ 
> > -   if ($input_errors) 
> > -       print_input_errors($input_errors);
> > -   else if (!$resolved && $type)
> > -       {
> > -       $errors[] = gettext("Host") . ": " . "\"" . $host . "\"" . " " . gettext("could not be resolved");
> > -       print_errors($errors, gettext("DNS Lookup error:"));  
> > -   }
> >   +?>
> > -  
> >   +<form action="diag_dns.php" method="post" name="iform" id="iform">
> >   +
> >   +<!-- First table displayes the DNS lookup form and the resulting IP addresses -->
> > -   <div class="table-responsive">
> >   I think this setup should be refactored; we don't need tables for layout. I think you should use panels for this; have a look at http://getbootstrap.com/components/#panels http://getbootstrap.com/components/#panels
> >   —
> >   Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/42/files#r27345696.

SjonHortensius replied:
I understand it's a bit more work; but if you get a good setup going it can be used for most diag_\* pages, let us know if you need any help!

sbeaver-netgate replied:
OK - Thanks :)

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com | www.netgate.com 

> On Mar 28, 2015, at 11:20 AM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/diag_dns.php https://github.com/SjonHortensius/pfsense/pull/42#discussion_r27345724:
> 
> > -                           echo $hostitem . "<br />";
> > -                           $found++;
> > -   /\* Dosplay any error messages resulting from user input */ 
> > -   if ($input_errors) 
> > -       print_input_errors($input_errors);
> > -   else if (!$resolved && $type)
> > -       {
> > -       $errors[] = gettext("Host") . ": " . "\"" . $host . "\"" . " " . gettext("could not be resolved");
> > -       print_errors($errors, gettext("DNS Lookup error:"));  
> > -   }
> >   +?>
> > -  
> >   +<form action="diag_dns.php" method="post" name="iform" id="iform">
> >   +
> >   +<!-- First table displayes the DNS lookup form and the resulting IP addresses -->
> > -   <div class="table-responsive">
> >   I understand it's a bit more work; but if you get a good setup going it can be used for most diag_\* pages, let us know if you need any help!
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/42/files#r27345724.

sbeaver-netgate committed:
Allow "Save" button to be suppressed

new Form(false) suppresses the “Save” button so that you can then add
your won custom button.

sbeaver-netgate committed:
Add a table to a section by passing it a two-dimensional array

Is there a better way to do this? Can the table position be controlled?

sbeaver-netgate committed:
Converted to bootstrap

HTML largely replaced by SH framework

sbeaver-netgate committed:
Updated conversion progress

Now based on ‘grep Form.class’ since some pages had foot.inc but had
not actually been converted.
